### ISSUE TITLE
Using query for all API Endpoints

### ISSUE THREAD

**Issue by charlesmudy** [knadh/listmonk#484]
**Is your feature request related to a problem? Please describe.**
Hello,

I'm using Listmonk for multiple users within a team/organisation to manage their campaigns and its a bit hard to customise it as I have to create external DB to manage the users. I was able to use create for subscribers which is good because it allows for extra field called attribs which I then used to add extra data for query

**Describe the solution you'd like**
Ideally, for such advanced customisation, it would be great to same attribs for campaigns (and all endpoints) and the possibility to use query on all... as it is currently designed for GET all campaigns specific to a user.

This might be silly, but how do I update our listmonk? Looking at some of the updates you have on master? without compromising what we have set-up already

Thanks in advance

**Comment by sweetppro** [comment]
for upgrading you only need to replace the binary, and run the upgrade script.
this is what I do on Ubuntu
```
cd /listmonk
rm listmonk
wget https://github.com/knadh/listmonk/releases/download/v2.0.0/listmonk_2.0.0_linux_amd64.tar.gz
tar -xzf listmonk_2.0.0_linux_amd64.tar.gz
rm listmonk_2.0.0_linux_amd64.tar.gz
./listmonk --upgrade
```

**Comment by mr-karan** [comment]
Please refer to [Release Notes](https://github.com/knadh/listmonk/releases) as they contain the Upgrade Instructions for both Docker/Binary and they might contain additional details about some breaking changes as well.

**Comment by charlesmudy** [comment]
Thanks everyone @mr-karan @sweetppro , I have done the migration successfully ... amazing work.

Still waiting on the query thing?

**Comment by knadh** [comment]
>  Ideally, for such advanced customisation, it would be great to same attribs for campaigns (and all endpoints) and the possibility to use query on all... as it is currently designed for GET all campaigns specific to a user.

hm, could you elaborate a bit on this? listmonk has no concept of per user campaigns.

**Comment by charlesmudy** [comment]
> > Ideally, for such advanced customisation, it would be great to same attribs for campaigns (and all endpoints) and the possibility to use query on all... as it is currently designed for GET all campaigns specific to a user.
> 
> hm, could you elaborate a bit on this? listmonk has no concept of per user campaigns.

`
const res = await axios({
				method: 'get',
				auth: {
					username: listmonkU,
					password: listmonkP
				},
				url: 'https://mail.domain.com/api/subscribers',
				params: {
					query: "subscribers.attribs->>'subscriber' = '" + userId + "'"
				}
			});
`

I am doing the above because I needed to assign group of subscriber to a particular admin. Basically we are using Listmonk to manage multiple subscribers.

I understand it doesn't have this, just saying if there is any chance to add additional attribute such as the above for each endpoint, it would make it scalable to have custom queries.

Currently, I have to forcefully do this: `var results = listmonkResult.data.data['results'].filter( i => firebaseResult.includes( i.id ) );` to ids in my DB to lismonks id for the records.

**Comment by knadh** [comment]
We manage our CRM integration at work by directly running queries on the Postgres DB. The listmonk table schema is really simple to work with. In fact, this is the recommended way to build bulk-integrations with external systems.

**Comment by charlesmudy** [comment]
That makes sense, will build on it then. Thanks.

One last thing, I don't want to open another issue: How can we add `{"Reply-To": "your@reply.com"}` per campaign?

**Comment by knadh** [comment]
> How can we add {"Reply-To": "your@reply.com"} per campaign?

Custom mail headers are only available per SMTP and not per campaign right now. I guess it makes sense to have this per-campaign. Will work on it.
